packages/mdc-select/_mixins.scss

```diff
@@ -0,0 +1,119 @@
+//
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Public
+
+@mixin mdc-select-ink-color($color) {
+  :not(.mdc-select--disabled) {
+    @include mdc-select-ink-color_($color);
+  }
+}
+
+@mixin mdc-select-fill-color($color) {
+  :not(.mdc-select--disabled) {
+    @include mdc-select-fill-color_($color);
+  }
+}
+
+@mixin mdc-select-label-color($color) {
+  &:not(.mdc-select--disabled) {
+    @include mdc-select-label-color_($color);
+  }
+}
+
+@mixin mdc-select-focused-label-color($color, $opacity: .87) {
+  &:not(.mdc-select--disabled) {
+    @include mdc-select-focused-label-color_($color, $opacity);
+  }
+}
+
+@mixin mdc-select-bottom-line-color($color) {
+  &:not(.mdc-select--disabled) {
+    @include mdc-select-bottom-line-color_($color);
+  }
+}
+
+@mixin mdc-select-focused-bottom-line-color($color) {
+  &:not(.mdc-select--disabled) {
+    @include mdc-select-focused-bottom-line-color_($color);
+  }
+}
+
+@mixin mdc-select-selected-text-color($color) {
+  &:not(.mdc-select--disabled) {
+    @include mdc-select-selected-text-color_($color);
+  }
+}
+
+// Private
+
+@mixin mdc-select-ink-color_($color) {
+  .mdc-select__surface {
+    @include mdc-theme-prop(color, $color);
+  }
+}
+
+@mixin mdc-select-fill-color_($color) {
+  .mdc-select__surface {
+    @include mdc-theme-prop(background-color, $color);
+  }
+}
+
+@mixin mdc-select-label-color_($color) {
+  .mdc-select__label {
+    @include mdc-theme-prop(color, $color);
+  }
+}
+
+@mixin mdc-select-focused-label-color_($color, $opacity: .87) {
+  .mdc-select__surface:focus .mdc-select__label {
+    @include mdc-theme-prop(color, $color);
+  }
+
+  // Separate parameter is used for opacity, because opacity is only applied when the
+  // label is floating, but the label is the same color when the select is focused
+  // but an option has not been selected.
+  .mdc-select__surface:focus .mdc-select__label--float-above {
+    opacity: $opacity;
+  }
+}
+
+@mixin mdc-select-bottom-line-color_($color) {
+  .mdc-select__bottom-line {
+    @include mdc-theme-prop(background-color, $color);
+  }
+}
+
+@mixin mdc-select-focused-bottom-line-color_($color) {
+  .mdc-select__surface:focus .mdc-select__bottom-line,
+  .mdc-select__surface:focus ~ .mdc-select__bottom-line {
```

Comment:
why do you need both these selectors?

Reply:
JS version has bottom line as a child element. CSS version is a sibling element.